File: invesdwin-context-parent/invesdwin-context-jruby/src/test/java/de/invesdwin/context/jruby/tests/InputsAndResultsTestShort.rb
print("getShort")
if (defined?(getShort)).nil?
	throw "getShort already defined!"
getShort = putShort
getShortType = typeof getShort
print(getShortType)
print(getShort)
if(getShortType !== 'number')
	throw "getShort not number!"

print("getShortVector")
if (defined?(getShortVector)).nil?
	throw "getShortVector already defined!"
getShortVector = putShortVector
getShortVectorType = typeof getShortVector[0]
print(getShortVectorType)
print(getShortVector)
if(getShortVectorType !== 'number')
	throw "getShortVector not number!"

print("getShortVectorAsList")
if (defined?(getShortVectorAsList)).nil?
	throw "getShortVectorAsList already defined!"
getShortVectorAsList = putShortVectorAsList
getShortVectorAsListType = typeof getShortVectorAsList[0]
print(getShortVectorAsListType)
print(getShortVectorAsList)
if(getShortVectorAsListType !== 'number')
	throw "getShortVectorAsList not number!"

print("getShortMatrix")
if (defined?(getShortMatrix)).nil?
	throw "getShortMatrix already defined!"
getShortMatrix = putShortMatrix
getShortMatrixType = typeof getShortMatrix[0][0]
print(getShortMatrixType)
print(getShortMatrix)
if(getShortMatrixType !== 'number')
	throw "getShortMatrix not number!"

print("getShortMatrixAsList")
if (defined?(getShortMatrixAsList)).nil?
	throw "getShortMatrixAsList already defined!"
getShortMatrixAsList = putShortMatrixAsList
getShortMatrixAsListType = typeof getShortMatrixAsList[0][0]
print(getShortMatrixAsListType)
print(getShortMatrixAsList)
if(getShortMatrixAsListType !== 'number')
	throw "getShortMatrixAsList not number!"

File: invesdwin-context-parent/invesdwin-context-clojure/src/test/java/de/invesdwin/context/clojure/tests/InputsAndResultsTestShort.clj
(defmacro defined [& var]
    `(try 
        (or 
            (boolean (resolve ~@var)) 
            (not (nil? (eval ~@var)))
        ) (catch Exception e# false)
    )
)

(println "getShort")
(if (defined 'getShort)
	(throw (Exception. "getShort already defined!"))
)
(def getShort putShort)
(def getShortType (class getShort))
(println getShortType)
(println getShort)
(if-not (= getShortType Short)
	(throw (Exception. "getShort not Short!"))
)

(println "getShortVector")
(if (defined 'getShortVector)
	(throw (Exception. "getShortVector already defined!"))
)
(def getShortVector putShortVector)
(def getShortVectorType (class (get getShortVector 0)))
(println getShortVectorType)
(println getShortVector)
(if-not (= getShortVectorType Short)
	(throw (Exception. "getShortVector not Short!"))
)

(println "getShortVectorAsList")
(if (defined 'getShortVectorAsList)
	(throw (Exception. "getShortVectorAsList already defined!"))
)
(def getShortVectorAsList putShortVectorAsList)
(def getShortVectorAsListType (class (get getShortVectorAsList 0)))
(println getShortVectorAsListType)
(println getShortVectorAsList)
(if-not (= getShortVectorAsListType Short)
	(throw (Exception. "getShortVectorAsList not Short!"))
)

(println "getShortMatrix")
(if (defined 'getShortMatrix)
	(throw (Exception. "getShortMatrix already defined!"))
)
(def getShortMatrix putShortMatrix)
(def getShortMatrixType (class (get (get getShortMatrix 0) 0)))
(println getShortMatrixType)
(println getShortMatrix)
(if-not (= getShortMatrixType Short)
	(throw (Exception. "getShortMatrix not Short!"))
)

(println "getShortMatrixAsList")
(if (defined 'getShortMatrixAsList)
	(throw (Exception. "getShortMatrixAsList already defined!"))
)
(def getShortMatrixAsList putShortMatrixAsList)
(def getShortMatrixAsListType (class (get (get getShortMatrixAsList 0) 0)))
(println getShortMatrixAsListType)
(println getShortMatrixAsList)
(if-not (= getShortMatrixAsListType Short)
	(throw (Exception. "getShortMatrixAsList not Short!"))
)
